### PR TITLE
[Refactor] Reduced nesting in MoveEffectPhase

### PR DESCRIPTION
**Pull request by ElizaAlex**
## What are the changes?
I adjusted MoveEffectPhase.run(), particularly the section that handles applying attributes when a move is used, to heavily reduce the amount of nested indents, and hopefully improve readability and clarity. 

## Why am I doing these changes?
I was working on another PR that required changes to this function, and found the changed section of code very difficult to read and follow what was going on. 

Improved comments have been added in the time between me finishing my other changes, and submitting this PR, but I still think these changes are worthwhile.

## What did change?
All changes refer to specifically the section of MoveEffectPhase related to constructing the final promise in charge of applying Move or Ability Attribute effects that is then pushed into applyAttrs.

- Flattened out nested promises into chains whenever reasonable
- Extracted sections of code out into named and documented helper functions

## How to test the changes?
Functionality _should_ remain entirely unchanged.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)

### CODE REVIEW

File: src/phases.ts [hunk 1]
```diff
@@ -3294,6 +3255,116 @@ export class MoveEffectPhase extends PokemonPhase {
     super.end();
   }
 
+  /**
+   * Apply self-targeted effects that trigger POST_APPLY
+   * Step skipped if user is Charging
```
Comment (innerthunder) [inline src/phases.ts:3260]: Minor clarification: `chargeEffect` is for moves with effects that are meant to trigger on the "charging turn" of charge moves (e.g. the stat boost from [Meteor Beam](https://bulbapedia.bulbagarden.net/wiki/Meteor_Beam_(move))). The big nested chain of move effects atm only triggers on the second turn of charge moves, which is why most attributes are ignored when `chargeEffect` is `true`

File: src/phases.ts [hunk 2]
```diff
@@ -3294,6 +3255,116 @@ export class MoveEffectPhase extends PokemonPhase {
...
+   * @param user
```
Comment (innerthunder) [inline src/phases.ts:3261]: I'm assuming that filling out these params is a WIP, but ideally this should follow a format similar to
```ts
/**
 * @param user {@linkcode Pokemon} insert description about this param
 */
```

**Review by innerthunder — CHANGES_REQUESTED** [review]
Some nits, but I like the structural changes to the code overall.

**Review by innerthunder — APPROVED** [review]

**Comment by ElizaAlex** [comment]
Updated to support removal of Phases.ts

**Comment by DayKev** [comment]
Ooh wish I had seen this sooner, are you still around to update the PR?

**Comment by ElizaAlex** [comment]
> Ooh wish I had seen this sooner, are you still around to update the PR?
Have been a bit too busy to work on any new PRs lately, but can prob get around to updating this and #3655 some time in the next few days/coming week.

**Review by innerthunder — DISMISSED** [review]

**Review by DayKev — COMMENTED** [review]
https://github.com/ElizaAlex/pokerogue/pull/1

Thanks for not notifying me this was updated GitHub, very useful.

**Review by DayKev — APPROVED** [review]

File: src/phases/move-effect-phase.ts [hunk 3]
```diff
@@ -52,12 +94,12 @@ export class MoveEffectPhase extends PokemonPhase {
      * Does an effect from this move override other effects on this turn?
      * e.g. Charging moves (Fly, etc.) on their first turn of use.
```
Comment (innerthunder) [inline src/phases/move-effect-phase.ts:95]: oh, this should probably use a different example since charge moves were reworked lol

**Review by innerthunder — APPROVED** [review]

File: src/phases/move-effect-phase.ts (hunk 3 above)
Comment (Tempo-anon) [reply src/phases/move-effect-phase.ts:95]: Just make a new PR to update it :snoremax:

**Pull request merged**